config/initializers/objection-connect.ts

```diff
@@ -7,6 +7,9 @@ export default {
   before: 'define-orm-models',
   async initialize(application: Application) {
     assert(application.config.database && application.config.database.client, 'Looks like you are missing database configuration. Add it to config.database - see the knex docs for configuration details: http://knexjs.org/#Installation-client');
-    application.container.register('objection:knex', Knex(application.config.database));
+    application.container.register('objection:knex', Knex(application.config.database), {
```

Comment:
The default options are `singleton: true, instantiate: false` (https://github.com/denali-js/denali/blob/master/lib/metal/container.ts#L14). This is correct, you want both to be `false` here. The default options tell Denali to apply injections to the object you are registering. Setting both to false tells Denali to skip the injection step, which is what we want here since the raw Knex object we are registering will never have any injections directly on it.